tangelo/linq/simulator_base.py

```diff
@@ -231,6 +265,86 @@ def get_expectation_value(self, qubit_operator, state_prep_circuit, initial_stat
+            if state_prep_circuit.width < len(term):
+                raise ValueError(f'Term {term} requires more qubits than the circuit contains ({state_prep_circuit.width})')
+            if type(coef) in {complex, np.complex64, np.complex128}:
+                are_coefficients_real = False
+
+        # If the underlying operator is hermitian, expectation value is real and can be computed right away
+        if are_coefficients_real:
+            if self._noise_model or not self.statevector_available \
+                    or state_prep_circuit.is_mixed_state or state_prep_circuit.size == 0:
+                return self._get_variance_from_frequencies(qubit_operator, state_prep_circuit, initial_statevector=initial_statevector)
+            elif self.statevector_available:
+                return self._get_variance_from_statevector(qubit_operator, state_prep_circuit, initial_statevector=initial_statevector)
+
+        # Else, separate the operator into 2 hermitian operators, use linearity and call this function twice
+        else:
+            qb_op_real, qb_op_imag = QubitOperator(), QubitOperator()
+            for term, coef in qubit_operator.terms.items():
+                qb_op_real.terms[term], qb_op_imag.terms[term] = coef.real, coef.imag
+            qb_op_real.compress()
+            qb_op_imag.compress()
+            var_real = self.get_variance(qb_op_real, state_prep_circuit, initial_statevector=initial_statevector)
+            var_imag = self.get_variance(qb_op_imag, state_prep_circuit, initial_statevector=initial_statevector)
+            # https://en.wikipedia.org/wiki/Complex_random_variable#Variance_and_pseudo-variance
+            return var_real if (var_imag == 0.) else var_real + var_imag  # always non-negative real number
+
+    def get_standard_error(self, qubit_operator, state_prep_circuit, initial_statevector=None):
+        r"""Take as input a qubit operator H and a quantum circuit preparing a
+        state |\psi>. Return the standard error of <\psi | H | \psi>, e.g. sqrt(Var H / n_shots).
+
+        In the case of a noiseless simulation, if the target backend exposes the
+        statevector then it is used directly to compute standard error (zero), or
+        draw samples if required. In the case of a noisy simulator, or if the
+        statevector is not available on the target backend, individual shots
+        must be run and the workflow is akin to what we would expect from an
+        actual QPU.
+
+        Args:
+            qubit_operator(openfermion-style QubitOperator class): a qubit
+                operator.
+            state_prep_circuit: an abstract circuit used for state preparation.
+
+        Returns:
+            complex: The standard error of this operator with regards to the
+                state preparation.
+        """
+        variance = self.get_variance(qubit_operator, state_prep_circuit, initial_statevector)
+
+        if not self.n_shots:
```

Comment:
Python offers its own version of the "ternary if", which is pretty concise and natural to read:
`return np.sqrt(variance/self.n_shots) if self.n_shots else 0.`